### PR TITLE
Documented the way to add a custom KubernetesAuthStrategy

### PR DESCRIPTION
**Pull request by andmagom**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by jamieklassen — CHANGES_REQUESTED** [review]
Got through about half and suggested some ways to address the vale errors. Beyond the in-line discussions, you can get this new page to appear in the sidebar by adding an entry here:

https://github.com/backstage/backstage/blob/61c7686961b3116c0784e60bb0eab1980d742596/microsite/sidebars.json#L96-L107

File: docs/features/kubernetes/authenticationstrategy.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
+---
+id: authenticationstrategy
```
Comment (vinzscam) [inline docs/features/kubernetes/authenticationstrategy.md:2]:
```suggestion
id: authentication-strategies
```

**Review by vinzscam — COMMENTED** [review]
wow fantastico!

**Review by vinzscam — APPROVED** [review]
:shipit:

**Comment by andmagom** [comment]
It looks like the pipeline error is on a file that I didn't modify

**Comment by vinzscam** [comment]
> It looks like the pipeline error is on a file that I didn't modify

it's fine we'll fix it

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**